sbi/inference/snle/mnle.py

```diff
@@ -205,4 +206,6 @@ def _loss(self, theta: Tensor, x: Tensor) -> Tensor:
         Returns:
             Negative log prob.
         """
-        return -self._neural_net.log_prob(x, context=theta)
+        theta = reshape_to_batch_event(theta, event_shape=theta.shape[1:])
+        x = reshape_to_iid_batch_event(x, event_shape=self._x_shape[1:])
```

Comment:
`x` could be 2D with leading dim being iid no? 
```suggestion
        x = reshape_to_iid_batch_event(x, event_shape=self._x_shape[1:], leading_is_iid=True)
```

Reply:
No! As with SNLE, in `loss`, we have only one `x` per `theta` (only one simulation result per `theta`). Therefore, the correct shapes are `(1, batch, event_x)` and `(batch, event_theta)`